src/shared/qr.spec.ts

```diff
@@ -343,5 +349,70 @@ describe('outbreakHistory functions', () => {
     });
   });
 
+  describe('deduplicateMatches', () => {
+    const locationId = 'abc123';
+    const checkIn1: CheckInData = {
+      id: locationId,
+      name: 'Burgers',
+      address: '123 King St',
+      timestamp: new Date(2021, 1, 10, 12).getTime(),
+    };
+    const checkIn2: CheckInData = {
+      id: locationId,
+      name: 'Burgers',
+      address: '123 King St',
+      timestamp: new Date(2021, 1, 11, 12).getTime(),
+    };
+    const outbreakEvent1: OutbreakEvent = {
+      id: 'outbreakEvent1',
+      locationId,
+      startTime: new Date(2021, 1, 10).getTime(),
+      endTime: new Date(2021, 1, 11).getTime(),
+      severity: 2,
+    };
+    const outbreakEvent2: OutbreakEvent = {
+      id: 'outbreakEvent2',
+      locationId,
+      startTime: new Date(2021, 1, 10).getTime(),
+      endTime: new Date(2021, 1, 11).getTime(),
+      severity: 3,
+    };
+    const outbreakEvent3: OutbreakEvent = {
+      id: 'outbreakEvent2',
+      locationId,
+      startTime: new Date(2021, 1, 11).getTime(),
+      endTime: new Date(2021, 1, 12).getTime(),
+      severity: 3,
+    };
+
+    it('filters out duplicate matches with a lower severity', () => {
+      const match1: MatchData = {
+        timestamp: checkIn1.timestamp,
+        checkIn: checkIn1,
+        outbreakEvent: outbreakEvent1,
+      };
+      const match2: MatchData = {
+        timestamp: checkIn1.timestamp,
+        checkIn: checkIn1,
+        outbreakEvent: outbreakEvent2,
+      };
+      expect(deduplicateMatches([match1, match2])).toStrictEqual([match2]);
+    });
+
```

Comment:
add test for same severity, different end time

Reply:
I removed this behaviour because it was buggy - now if 2 overlapping scans have the same severity, it will just pick the first one it iterates over.